katdal/categorical.py

```diff
@@ -544,6 +582,9 @@ def sensor_to_categorical(sensor_timestamps, sensor_values, dump_midtimes, dump_
     sensor_timestamps = np.atleast_1d(sensor_timestamps)
     sensor_values = np.atleast_1d(sensor_values)
     dump_midtimes = np.atleast_1d(dump_midtimes)
+    # Check if sensor values are objects wrapped in HashableValueWrappers
+    wrapped_values = len(sensor_values) and isinstance(sensor_values[0],
+                                                       HashableValueWrapper)
```

Comment:
This function seems to be complicated by the need to do alternative versions of all the processing which does un-wrapping and rewrapping. Wouldn't it be simpler to do:
1. Unwrap if wrapped.
2. Do the operations on the raw values (avoiding putting them into numpy).
3. Rewrap if appropriate.